cl/corpus_importer/management/commands/harvard_opinions.py

```diff
@@ -255,14 +267,30 @@ def parse_harvard_opinions(reporter, volume, make_searchable):
             continue
 
         # TODO: Generalize this to handle all court types somehow.
-        court_id = match_court_string(
-            data["court"]["name"],
-            state=True,
-            federal_appeals=True,
-            federal_district=True,
+        if not options["court_id"]:
+            # Sometimes courts can't match just one court
+            # This is used to alleviate certain circumstances
+            court_id = match_court_string(
+                data["court"]["name"],
+                state=True,
+                federal_appeals=True,
+                federal_district=True,
+            )
+        # Handle partial dates by adding -01v to YYYY-MM dates
+        date_filed, is_approximate = validate_dt(data["decision_date"])
+        case_body = data["casebody"]["data"]
+
+        previously_imported_case = find_previously_imported_cases(
+            court_id, date_filed, case_body, data["docket_number"]
         )
+        if previously_imported_case:
+            with transaction.atomic():
+                add_citations(
+                    data["citations"], cluster_id=previously_imported_case.id
+                )
+            continue
 
-        soup = BeautifulSoup(data["casebody"]["data"], "lxml")
+        soup = BeautifulSoup(case_body, "lxml")
```

Comment:
A commend above this line saying, in effect, "We haven't seen this before, add it..." Would be helpful. Or, if you want to go bigger, you could pull the remaining code out of this function and put it in a `add_new_opinion` function or something. That'd be very clear, I think.

Reply:
√